### PR TITLE
feat(query): add ability to pass arrays as replacements #9050

### PR DESCRIPTION
**Pull request by Kirill89**
According to the [feature request](https://github.com/sequelize/sequelize/issues/9050) added ability to pass arrays to `sequelize.query()` as replacements.

### CODE REVIEW

File: lib/sql-string.js [hunk 1]
```diff
@@ -4,6 +4,23 @@ const dataTypes = require('./data-types');
 const util = require('util');
 const _ = require('lodash');
 
+function arrayToList(array, timeZone, dialect, format) {
```
Comment (sushantdhiman) [inline lib/sql-string.js]: Can you cleanup this method, may be use reduce as well

File: lib/sql-string.js [hunk 2]
```diff
@@ -44,7 +61,7 @@ function escape(val, timeZone, dialect, format) {
     if (dialect === 'postgres' && !format) {
       return dataTypes.ARRAY.prototype.stringify(val, {escape: partialEscape});
     }
-    return val.map(partialEscape);
```
Comment (sushantdhiman) [inline lib/sql-string.js]: I believe this will break other dialects, MSSQL i think. Will probably need more tests

File: lib/sql-string.js (hunk 1 above)
Comment (Kirill89) [reply lib/sql-string.js]: Done

File: lib/sql-string.js (hunk 2 above)
Comment (Kirill89) [reply lib/sql-string.js]: I added integration test, looks like it's okay with MSSQL.

**Comment by Kirill89** [comment]
@sushantdhiman , any updates so far?

File: lib/sql-string.js (hunk 2 above)
Comment (Vovka19) [reply lib/sql-string.js]: I will learn. didn't sleep 3 days. thank you

File: test/integration/sequelize.test.js [hunk 3]
```diff
@@ -273,6 +273,23 @@ describe(Support.getTestDialectTeaser('Sequelize'), () => {
       return this.sequelize.query(this.insertQuery);
     });
 
+    it('executes a query if a placeholder value is an array', function() {
```
Comment (eseliger) [inline test/integration/sequelize.test.js]: Maybe adding a unit test to check for the output of the query generator as well might be good?

Comment (Kirill89) [reply test/integration/sequelize.test.js]: @eseliger, done.

**Comment by Kirill89** [comment]
@sushantdhiman, @eseliger, any updates? I really need this feature.

**Pull request merged**